src/vconsole/vconsole-setup.c

```diff
@@ -223,51 +223,85 @@ static int font_load_and_wait(const char *vc, const char *font, const char *map,
  * processing, stty iutf8
  */
 static void setup_remaining_vcs(int fd, int vc_max, bool utf8) {
+        struct console_font_op cfo;
         struct vt_stat vcs = {};
+        struct unimapinit adv = {};
         struct unimapdesc unimapd;
         _cleanup_free_ struct unipair* unipairs = NULL;
-        int i, r;
+        _cleanup_free_ unsigned char *fontbuf = NULL;
+        const char *err_ioctl;
+        int i, r, fd_d;
 
         unipairs = new(struct unipair, USHRT_MAX);
         if (!unipairs) {
                 log_oom();
                 return;
         }
 
+        fontbuf = new(unsigned char, 32*32*512/8);
+        if (!fontbuf) {
+                log_oom();
+                return;
+        }
+
         /* get active, and 16 bit mask of used VT numbers */
         r = ioctl(fd, VT_GETSTATE, &vcs);
         if (r < 0) {
                 log_warning_errno(errno, "VT_GETSTATE failed, ignoring remaining consoles: %m");
                 return;
         }
 
-        for (i = 1; i <= vc_max; i++) {
-                int vcfd;
-                struct console_font_op cfo = {};
+        /* get fonts from source console */
+        cfo.op = KD_FONT_OP_GET;
+        cfo.flags = 0;
+        cfo.width = cfo.height = 32;
+        cfo.charcount = 512;
+        cfo.data = fontbuf;
```

Comment:
It would be nicer to set constant fields this during initialization:
```
struct console_font_op cfo {
    .op = KD_FONT_OP_GET,
    .width = .height = 32,
    .charcount = 512,
};
```